src/www/views/servers_view/server_connection_indicator/index.ts

```diff
@@ -11,284 +11,191 @@
-      /* rtl:end:ignore */
-    </style>
-    <div id="container" class$="[[expandedClassName]]">
-      <img id="small-grey" src$="[[rootPath]]assets/disc_grey.png" class$="grey {{animationState}}" />
-      <img id="small" src$="[[rootPath]]assets/disc_color.png" class$="green {{animationState}}" />
-      <img id="medium-grey" src$="[[rootPath]]assets/disc_grey.png" class$="grey {{animationState}}" />
-      <img id="medium" src$="[[rootPath]]assets/disc_color.png" class$="green {{animationState}}" />
-      <img id="large-grey" src$="[[rootPath]]assets/disc_grey.png" class$="grey {{animationState}}" />
-      <img id="large-zero" src$="[[rootPath]]assets/disc_empty.png" class$="{{state}}" />
-      <img id="large" src$="[[rootPath]]assets/disc_color.png" class$="green {{animationState}}" />
-    </div>
-  `;
-
-  animationStartMS: number;
-
-  @property({type: String}) rootPath: string;
-  @property({type: Boolean}) expanded: boolean;
-
-  @property({type: String, observer: "syncAnimationState"}) state: ServerConnectionState;
-  @property({type: String}) animationState: ServerConnectionState = ServerConnectionState.INITIAL;
-
-  @computed("expanded")
-  get expandedClassName() {
-    return this.expanded ? "expanded" : "";
-  }
 
-  @computed("state")
-  get shouldAnimate() {
-    return this.isAnimationState(this.state);
-  }
+        .circle-${circleSize}.circle-connecting,
+        .circle-${circleSize}.circle-reconnecting,
+        .circle-${circleSize}.circle-disconnecting {
+          animation-name: circle-${circleSize}-rotate-with-pause;
+        }
 
-  @computed("animationState")
-  get isAnimating() {
-    return this.isAnimationState(this.animationState);
-  }
+        /* Do not mirror animation for RTL languages */
+        /* rtl:begin:ignore */
+        @keyframes circle-${circleSize}-rotate-with-pause {
+          0% {
+            transform: rotate(0deg) var(--circle-${circleSize}-scale);
+          }
+          60%,
+          100% {
+            transform: rotate(360deg) var(--circle-${circleSize}-scale);
+          }
+        }
+        /* rtl:end:ignore */
+      `
+    ),
+  ];
 
-  // @ts-ignore
-  private syncAnimationState() {
+  render() {
     if (this.shouldAnimate) {
-      return this.startAnimation();
+      this.startAnimation();
+    } else if (this.isAnimating && !this.shouldAnimate) {
+      this.stopAnimation();
+    } else {
+      this.animationState = this.connectionState;
     }
 
-    if (!this.shouldAnimate && this.isAnimating) {
-      return this.stopAnimation();
-    }
+    const circles = this.animationState === ServerConnectionState.INITIAL ? [CIRCLE_SIZES[0]] : CIRCLE_SIZES;
+
+    return html`
+      ${circles.map(
+        circleSize =>
+          html`
+            <img
+              class="circle circle-${circleSize} circle-${this.animationState}"
+              src="${this.rootPath}assets/circle.png"
+            />
+          `
+      )}
+    `;
+  }
+
+  private get shouldAnimate() {
```

Comment:
I'd get rid of this function to remove unnecessary indirection and improve readability.
With Lit you can inline `this.isAnimationState(this.connectionState)`